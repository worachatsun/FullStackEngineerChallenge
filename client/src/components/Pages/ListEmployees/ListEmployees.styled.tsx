import styled from 'styled-components';

export const FlexAlignCenter = styled.div`
  display: flex;
  align-items: center;
`;

export const Container = styled(FlexAlignCenter)`
  width: 100%;
  flex-direction: column;
`;

export const Wrapper = styled(FlexAlignCenter)`
  width: 70%;
  flex-direction: column;
`;

export const ListHeader = styled(FlexAlignCenter)`
  flex-direction: row;
  width: 100%;
  justify-content: space-around;
`;
