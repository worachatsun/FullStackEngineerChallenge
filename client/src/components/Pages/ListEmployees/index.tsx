import React, { FunctionComponent } from "react";
import { FaUserTie } from "react-icons/fa";
import useSWR from "swr";
import { LIST_EMPLOYEES_API } from "../../../constants/routes";
import { IListModel } from "../../../interfaces/model";
import Button from "../../commons/Button";
import Empty from "../../commons/Empty";
import Modal from "../../commons/Modal";
import { fetcher } from "../../commons/utils/client";
import useModal from "../../Hook/useModal";
import Layout from "../../Layout";
import Assign from "../Assign";
import Remove from "../Auth/Remove";
import Signup from "../Auth/Signup";
import { ButtonContainer, Hr, Lable, ListHeader, Table, Th, ThText, Tr } from "./ListEmployees.styled";

const ListEmployees: FunctionComponent = () => {
    const { isShowing, toggle, close } = useModal();
    const token = localStorage.getItem("token");
    const { data, mutate } = useSWR<IListModel[]>([LIST_EMPLOYEES_API, token as string], fetcher);

    return data ? (
        <Layout>
            <ListHeader>
                <Lable>List Employees</Lable>
                <Button onClick={toggle}>Add Employee</Button>
            </ListHeader>
            <Table>
                <Th>
                    <FaUserTie size={"1.5em"} color="#ff0033" />
                    <ThText>Name</ThText>
                </Th>
                {data.map((user: IListModel, index: number) => (
                    <div key={index}>
                        <Hr />
                        <Tr>
                            <div>{user.username}</div>
                            <ButtonContainer>
                                <Assign username={user.username} users={data} token={token as string} />
                                <Remove username={user.username} mutate={mutate} />
                            </ButtonContainer>
                        </Tr>
                    </div>
                ))}
                <Hr />
                {data.length === 0 ? <Empty title="No user yet. Please add a user." /> : <div />}
            </Table>
            <Modal isShowing={isShowing} close={close}>
                <Signup />
            </Modal>
        </Layout>
    ) : (
        <div>Loading</div>
    );
};

export default ListEmployees;
