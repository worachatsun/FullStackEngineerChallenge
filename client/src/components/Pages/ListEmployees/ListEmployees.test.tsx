import React from "react";
import renderer from "react-test-renderer";
import ListEmployees from "./index";

describe("<ListEmployees />", () => {
    test("should render correctly", () => {
        const render = renderer.create(<ListEmployees />).toJSON();
        expect(render).toMatchSnapshot();
    });
});
