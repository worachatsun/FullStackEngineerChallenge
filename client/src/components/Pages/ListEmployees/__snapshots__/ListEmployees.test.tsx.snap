// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> should render correctly 1`] = `<div />`;

exports[`<ListEmployees /> should render correctly 1`] = `<div />`;
